### PR TITLE
fix: Invalid Date object will throw a TypeError (#58)

### PR DESCRIPTION
**Pull request by dotnetCarpenter**
This PR will make sure that `loupe` will return `"Invalid Date"` if a `Date` object is invalid. It does **not** deal with _how_ the `Date` object became invalid as that depends on JS engine.

See https://github.com/chaijs/loupe/issues/58#issuecomment-1303259185 for a comparison between V8 and Firefox's SpiderMoney.

### CODE REVIEW

File: lib/date.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import { truncate } from './helpers'
 
 export default function inspectDate(dateObject, options) {
-  // If we need to - truncate the time portion, but never the date
-  const split = dateObject.toJSON().split('T')
-  const date = split[0]
-  return options.stylize(`${date}T${truncate(split[1], options.truncate - date.length - 1)}`, 'date')
+  const stringRepresentation = dateObject.toJSON()
+  if (stringRepresentation === null) {
+    return 'Invalid Date'
```
Comment (keithamus) [inline lib/date.js:6]: Do you think checking `NaN` here would be better?

```suggestion
  if (Number.isNaN(dateObject.valueOf())) {
    return 'Invalid Date'
```

Comment (dotnetCarpenter) [reply lib/date.js:6]: Good question. In my test both SpiderMonkey and V8 returns `null` for `.toJSON()`, so calling `Number.isNaN` is just a needless extra call. But if there are implementations out there that returns anything else when `.toJSON()` is called on an invalid `Date` object, then yes - `NaN` is better.

However, [21.4.4.37 Date.prototype.toJSON ( key )](https://262.ecma-international.org/#sec-date.prototype.tojson) might be doing those steps (`Number.isNaN(dateObject.valueOf())`) anyway. Step 2 is _"Let tv be ? [ToPrimitive](https://262.ecma-international.org/#sec-toprimitive)(O, number)"_ which will call `.valueOf()` but step 3 confuses me.

Comment (dotnetCarpenter) [reply lib/date.js:6]: `dateObject.valueOf()` will always return a `Number` for a `Date` object - `NaN` is also a `Number` type.

1. `dateObject.toJSON()` calls `valueOf()` and checks that, a `Number` is finite - so that is an extra check we get for free*.
    1. `Number.isFinite(NaN)` and `Number.isFinite(9e999)` are both `false` for example.
1. It then calls `toISOString()` which always throws a `RangeError: invalid date`, if somehow the above check was not enough and will return `null` in that case.

_\* since we need to call `.toJSON()` anyway to get the serialization to work_

In short, I don't think `Number.isNaN(dateObject.valueOf())` is better than `dateObject.toJSON() === null`, since we have to do the latter anyway.

**Comment by dotnetCarpenter** [comment]
@keithamus Any reason to why the CI stuff is not running?

**Comment by keithamus** [comment]
I needed to approve it for running. It's running now

**Comment by dotnetCarpenter** [comment]
I also ran the Karma test suite against IE 11 on Windows 10 successfully.

**Review by keithamus — APPROVED** [review]
